Solutions/Corvus.Extensions.CosmosClient/Corvus/Extensions/Cosmos/EntityInstance{T}.cs

```diff
@@ -16,20 +16,14 @@ namespace Corvus.Extensions.Cosmos
     /// <remarks>This provides a serializable version of the response which decorates the entity with an <see cref="ETag"/>.</remarks>
     [JsonConverter(typeof(EntityInstanceJsonConverter))]
     public sealed class EntityInstance<T> : IEquatable<IEntityInstance<T>>, IEntityInstance<T>
+        where T : class
```

Comment:
Is this valid, or could we expect a `struct`?

Reply:
Without the constraint:
![image](https://user-images.githubusercontent.com/2104606/79359601-b46b8f00-7f3a-11ea-9792-bdc9ec15387a.png)
Maybe @mwadams could answer?

Reply:
We absolutely need to support structs here. This is a wrapper for a type that decorates it with ETag and Timestamp and is used to return the struct from a repo. We don't want it then doing all the boxing/unboxing stuff.

And that is why you are finding it tricky with nullability!

Reply:
Looking at it, that should be returning object? As it is nullable.

Reply:
Returning `object?` has fixed the issue

Reply:
OK, but an alternative you might try is a `notnull` constraint on `IEntity<T>`, which will also mean you'll need to express the same constraint in this class. As I understand it, that lets you state "I don't care whether this is a struct or a class, but I definitely want it to be not null". This would enable `IEntity<T>.Entity` to be a non-null `object`.

And I think that's what you want isn't it? There's no valid situation in which `Entity` should be able to return `null` right? So making this `object?` in order just to fix the warning is a move in the wrong direction isn't it?

Reply:
BTW, I have a series of blog posts coming up all about these annoying little corner cases where nullability meets generics.

Reply:
I assumed @mwadams was suggesting that the `Entity` *could* be `null` here?

Reply:
Hitherto, it could be... Whether it should be? I think it would be reasonable to *not* support null. And in fact it might be better.

Reply:
> OK, but an alternative you might try is a `notnull` constraint on `IEntity<T>`, which will also mean you'll need to express the same constraint in this class.

These attributes are only available in netstandard2.1+

Do we want to update this project to 2.1 to support this, or alternatively we could add the attribute definition to the project (as per [the suggestion in this blog post](https://www.thinktecture.com/en/net-core/c-8-nullable-reference-types-in-net-standard-21-and-net-4x/)?

Reply:
Well, this is an interesting question. Because hitherto we have left Corvus at netstandard2.0 for maximum usability. Change to netstandard2.1 would be a lot of breaking changes.

@idg10?

Reply:
Also someone has created a compatibility pack and pushed to NuGet, that we could use if we don't want to upgrade
https://github.com/Therzok/nullability-compat

Reply:
We've been using 'Nullable' everywhere else for this. https://www.nuget.org/packages/Nullable/

Reply:
(Which is the supported package, isn't it?)

Reply:
Turns out the package wasn't required. I thought we would need to use the nullable attributes, but that's not the case. Updated now with `notnull` constraint.